packages/frontend/src/components/DashboardFilter/FilterConfiguration/utils/index.ts

```diff
@@ -35,10 +35,11 @@ export const hasFilterValueSet = (filterRule: DashboardFilterRule) => {
         case FilterOperator.NOT_IN_THE_CURRENT:
             return filterRule.settings && filterRule.settings.unitOfTime;
         case FilterOperator.IN_BETWEEN:
+        case FilterOperator.NOT_IN_BETWEEN:
             return (
                 filterRule.values &&
                 filterRule.values.length === 2 &&
-                filterRule.values.every(Boolean)
+                filterRule.values.every((val) => val != null && val !== '')
```

Comment:
For just dates `Boolean` would've worked, but we need to properly check for `0` now for numers